local-cluster/tests/local_cluster.rs

```diff
@@ -1536,20 +1538,42 @@ fn test_wait_for_max_stake() {
     solana_logger::setup_with_default(RUST_LOG_FILTER);
     let validator_config = ValidatorConfig::default_for_test();
     let slots_per_epoch = MINIMUM_SLOTS_PER_EPOCH;
+    // Set this large enough to allow for skipped slots but still be able to
+    // make a root and derive the new leader schedule in time.
+    let stakers_slot_offset = slots_per_epoch.saturating_mul(MAX_LEADER_SCHEDULE_EPOCH_OFFSET);
+    // Reduce this so that we can complete the test faster by advancing through
+    // slots/epochs faster. But don't make it too small because it can make us
+    // susceptible to skipped slots and the cluster getting stuck.
+    let ticks_per_slot = 16;
     let mut config = ClusterConfig {
         cluster_lamports: DEFAULT_CLUSTER_LAMPORTS,
         node_stakes: vec![DEFAULT_NODE_STAKE; 4],
         validator_configs: make_identical_validator_configs(&validator_config, 4),
         slots_per_epoch,
-        stakers_slot_offset: slots_per_epoch,
+        stakers_slot_offset,
+        ticks_per_slot,
         ..ClusterConfig::default()
     };
     let cluster = LocalCluster::new(&mut config, SocketAddrSpace::Unspecified);
     let client = RpcClient::new_socket(cluster.entry_point_info.rpc().unwrap());
 
-    assert!(client
-        .wait_for_max_stake(CommitmentConfig::default(), 33.0f32)
-        .is_ok());
```

Comment:
nit: What was the reason for shifting from `assert!` to `if let Err()` + `panic!`; wanting to see the error ? If so, the following might also work:
```rust
assert_eq!(Ok(), client.wait_for_max_stake_below_threshold_with_timeout(...))
```
I'd be happy shipping this PR with no changes here tho

Reply:
Yes, just trying to see the error.

I actually tried what you're suggesting first and got hit with this:
```
can't compare `solana_rpc_client_api::client_error::Error` with `solana_rpc_client_api::client_error::Error`
the trait `PartialEq` is not implemented for `solana_rpc_client_api::client_error::Error`, which is required by `Result<(), _>: PartialEq<_>`
required for `Result<(), solana_rpc_client_api::client_error::Error>` to implement `PartialEq`
```